fuel/app/classes/controller/api/asset.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Materia
+ * License outlined in licenses folder
+ */
+
+use \Materia\Perm;
+use \Materia\Perm_Manager;
+
+class Controller_Api_Asset extends Controller_Rest
+{
+
+	public function post_delete($asset_id)
+	{
+		$user_id = \Model_User::find_current_id();
+		if (\Service_User::verify_session() !== true) return Msg::no_login();
+
+		if ( ! Perm_Manager::user_has_any_perm_to($user_id, $asset_id, Perm::ASSET, Perm::FULL))
+			return new Response('You don not have access to this asset', 401);
+
+		try
+		{
+			\DB::update('asset')
+				->value('deleted_at', time())
+				->value('is_deleted', '1')
+				->where('id', $asset_id)
+				->execute();
+		}
+		catch (\Exception $th)
+		{
+			trace('Error: In the deletion process');
+			trace($th);
```

Comment:
Not sure if we should return an error message here.

Reply:
You're right, the request will now return an error message consistent with other API requests.